t/db/setup_test_db.pl

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+setup_test_db -
+
+=cut
+
+=head1 DESCRIPTION
+
+setup_test_db
+
+=cut
+
+use strict;
+use warnings;
+use lib qw(/usr/local/pf/lib);
+BEGIN {
+    use lib '/usr/local/pf/t';
+    use setup_test_config;
+}
+
+use pf::db;
+use DBI;
+
+my $config = pf::db::db_config();
+
+if ($config->{user} ne 'pf_smoke_tester') {
+   die "Not using the standard testing user for db\n"; 
+}
+
+if ($config->{db} !~ /^pf_smoke_test/) {
+   die "Not using the standard database for testing \n";
+}
+
+my $dbh = test_db_connection($config) or die <<"EOS";
+Cannot connection to db with test user please run
+mysql -uroot -p < /usr/local/pf/t/db/smoke_test.sql;
+EOS
+my $db = $config->{db};
+
+$dbh->do("DROP DATABASE IF EXISTS $db;") or die "Cannot drop database $db\n";;
+$dbh->do("CREATE DATABASE $db;") or die "Cannot create database $db\n";
+system("mysql -u$config->{user} -p$config->{pass} $db < /usr/local/pf/db/pf-schema-X.Y.Z.sql");
+
+sub test_db_connection {
+    my ($config) = @_;
+    my $dsn = "dbi:mysql:;host=$config->{host};port=$config->{port};mysql_client_found_rows=0";
+    return  DBI->connect($dsn, $config->{user}, $config->{pass}, { RaiseError => 0, PrintError => 0, mysql_auto_reconnect => 1 }); 
```

Comment:
Shouldn't we connect with the same parameters as in pf::db ?

Maybe a quick util function to create the DB connection would be useful ?

Reply:
I am connecting without a database.

Reply:
This is a special case where I want to connect to mysql without setting the current database.
Because if I try and connect with a current database that does exists it will fail.
So no need for a util function at the moment.
If we need it later I will add it.